include/nv/detail/__target_macros

```diff
@@ -0,0 +1,395 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _NV__TARGET_MACROS
+#define _NV__TARGET_MACROS
+
+#include "__preprocessor"
+
+#if defined(_NV_COMPILER_NVCXX)
+
+#  define _NV_TARGET_VAL_SM_35 nv::target::sm_35
+#  define _NV_TARGET_VAL_SM_37 nv::target::sm_37
+#  define _NV_TARGET_VAL_SM_50 nv::target::sm_50
+#  define _NV_TARGET_VAL_SM_52 nv::target::sm_52
+#  define _NV_TARGET_VAL_SM_53 nv::target::sm_53
+#  define _NV_TARGET_VAL_SM_60 nv::target::sm_60
+#  define _NV_TARGET_VAL_SM_61 nv::target::sm_61
+#  define _NV_TARGET_VAL_SM_62 nv::target::sm_62
+#  define _NV_TARGET_VAL_SM_70 nv::target::sm_70
+#  define _NV_TARGET_VAL_SM_72 nv::target::sm_72
+#  define _NV_TARGET_VAL_SM_75 nv::target::sm_75
+#  define _NV_TARGET_VAL_SM_80 nv::target::sm_80
+#  define _NV_TARGET_VAL_SM_86 nv::target::sm_86
+
+#  define _NV_TARGET_VAL nv::target::device
+
+#  define _NV_TARGET___NV_IS_HOST nv::target::is_host
+#  define _NV_TARGET___NV_IS_DEVICE nv::target::is_device
+
+#  define _NV_TARGET___NV_ANY_TARGET (nv::target::is_host || nv::target::is_device)
+#  define _NV_TARGET___NV_NO_TARGET !_NV_TARGET___NV_ANY_TARGET
+
+#if defined(__NVCOMPILER_CUDA_ARCH__)
```

Comment:
`__NVCOMPILER_CUDA_ARCH__` will go away once Allison finishes refactoring CUB.  It shouldn't be referenced at all by `<nv/target>`.  I think `_NV_TARGET_DEF` is not necessary.  `_NV_TARGET_MINIMUM_VAL` should be kept, but on its own and not within a `#if defined(__NVCOMPILER_CUDA_ARCH__)` block.

Reply:
I don't see it being used in the code base at this point